hooks/VDM.js

```diff
@@ -268,19 +304,304 @@ module.exports = function (input, session) {
+        path: `environment.observations.${meteoLocation}.wave.significantHeight`,
+        value: data.signwavewhgt / 10,
+      })
+    }
+  }
+
+  if (data.waveperiod) {
+    contextPrefix = 'meteo.'
+    if (data.waveperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.period`,
+        value: data.waveperiod,
+      })
+    }
+  }
+
+  if (data.wavedir) {
+    contextPrefix = 'meteo.'
+    if (data.wavedir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.direction`,
+        value: utils.transform(data.wavedir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.swellhgt) {
+    contextPrefix = 'meteo.'
+    if (data.swellhgt < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.height`,
+        value: data.swellhgt / 10,
+      })
+    }
+  }
+
+  if (data.swellperiod) {
+    contextPrefix = 'meteo.'
+    if (data.swellperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.period`,
+        value: data.swellperiod,
+      })
+    }
+  }
+
+  if (data.swelldir) {
+    contextPrefix = 'meteo.'
+    if (data.swelldir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.direction`,
+        value: utils.transform(data.swelldir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.seastate) {
+    contextPrefix = 'meteo.'
+    if (data.seastate < 13) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.seaState`,
+        value: beaufortScale[data.seastate],
+      })
+    }
+  }
+
+  if (data.watertemp) {
+    contextPrefix = 'meteo.'
+    if (data.watertemp > -101 && data.watertemp < 501) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.temperature`,
+        value: utils.transform((data.watertemp / 10), 'c', 'k'),
+      })
+    }
+  }
+
+  if (data.precipitation) {
+    contextPrefix = 'meteo.'
+    if (data.precipitation < 7) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.precipitation`,
+        value: precipitationType[data.precipitation],
+      })
+    }
+  }
+
+  if (data.salinity) {
+    contextPrefix = 'meteo.'
+    if (data.salinity < 502) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.salinity`,
+        value: data.salinity / 10,
+      })
+    }
+  }
+
+  if (data.ice) {
+    contextPrefix = 'meteo.'
+    if (data.ice < 3) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.ice`,
+        value: iceTable[data.ice],
+      })
+    }
+  }
+
   if (values.length === 0) {
     return null
   }
 
-  const delta = {
-    context: contextPrefix + `urn:mrn:imo:mmsi:${data.mmsi}`,
-    updates: [
-      {
-        source: tags.source,
-        timestamp: tags.timestamp,
-        values: values,
-      },
-    ],
+  if (contextPrefix === 'meteo.'){
```

Comment:
To be consistent with all other contexts I think that `meteo.{identifier}` should be used.

Reply:
I did got the idea here. Can you explain more or can you propose how it should be?

All weather data is labelled with `meteo.` prefix, making context e.g
"context": "meteo.urn:mrn:imo:mmsi:002655619"
This just pick those for special threatment.

Reply:
If "<lat>_<lon>" is the differentiator then rather in the path can it be made part of the identifier rather than in the path?

"context": "meteo.urn:mrn:imo:mmsi:<xxxxxxxxx>_<lat>_<lon>"

From a client perspective the identifier contains all the data required to id the resource and can easily "split" the value using "_".

Reply:
> \"context

This returning back to making fake mssi to populate each weather station in own context. To my understanding, this was not the wanted way.